src/server/campaignd/server.cpp

```diff
@@ -898,30 +1065,113 @@ void server::handle_upload(const server::request& req)
 		}
 
 		const std::string& filename = (*campaign)["filename"].str();
-		data["title"] = (*campaign)["title"];
 		data["name"] = "";
-		data["campaign_name"] = (*campaign)["name"];
-		data["author"] = (*campaign)["author"];
-		data["description"] = (*campaign)["description"];
-		data["version"] = (*campaign)["version"];
-		data["timestamp"] = (*campaign)["timestamp"];
-		data["original_timestamp"] = (*campaign)["original_timestamp"];
-		data["icon"] = (*campaign)["icon"];
-		data["type"] = (*campaign)["type"];
-		data["tags"] = (*campaign)["tags"];
 		find_translations(data, *campaign);
 
 		add_license(data);
+		
+		const std::string& new_version = (*campaign)["version"].str();
+		//#TODO: probably make hash from new_version + required_wesnoth_version ?
+		const std::string& file_hash = utils::md5(new_version).hex_digest();
```

Comment:
Why is the hash used here calculated from the version number instead of the archive contents?

Reply:
I rescind the previous comment. Obviously we can't use unsanitized input to locate files because otherwise we could get into serious trouble.